File: library/src/main/java/ga/rugal/git/model/GitPackIndexEntry.java
package ga.rugal.git.model;

import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.RequiredArgsConstructor;
import lombok.Setter;
import lombok.ToString;

/**
 * Pack index entry.
 *
 * @author Rugal Bernstein
 */
@EqualsAndHashCode(onlyExplicitlyIncluded = true)
@Getter
@RequiredArgsConstructor
@SuppressFBWarnings({"EI_EXPOSE_REP", "EI_EXPOSE_REP2"})
@ToString
public class GitPackIndexEntry {

  /**
   * Blob id.
   */
  private final String id;

  /**
   * Byte offset notated in pack index file.
   */
  private final long offset;

  /**
   * File size after compression.
   */
  @Setter
  private long size;
}

File: library/src/main/kotlin/ga/rugal/gitcleaner/core/model/GitPackIndexEntry.kt
package ga.rugal.gitcleaner.core.model

/**
 * Pack index entry, without path.
 *
 * @author Rugal Bernstein
 */
data class GitPackIndexEntry(
  /**
   * SHA-1 of the git object: blob.
   */
  val id: String,
  /**
   * File size byte after compression, it is offset difference.
   */
  val size: Long,
) {
  override fun equals(other: Any?): Boolean {
    if (this === other) return true
    if (javaClass != other?.javaClass) return false

    other as GitPackIndexEntry

    return id == other.id
  }

  override fun hashCode(): Int = id.hashCode()
}
